src/lib.rs

```diff
@@ -17,19 +17,30 @@ mod mbr;
 mod pxe;
 
 const KERNEL_FILE_NAME: &str = "kernel-x86_64";
+const RAMDISK_FILE_NAME: &str = "ramdisk-x86_64";
 const BIOS_STAGE_3: &str = "boot-stage-3";
 const BIOS_STAGE_4: &str = "boot-stage-4";
 
 /// Create disk images for booting on legacy BIOS systems.
 pub struct BiosBoot {
     kernel: PathBuf,
+    ramdisk: Option<PathBuf>,
 }
 
 impl BiosBoot {
     /// Start creating a disk image for the given bootloader ELF executable.
     pub fn new(kernel_path: &Path) -> Self {
         Self {
             kernel: kernel_path.to_owned(),
+            ramdisk: None,
+        }
+    }
+
+    /// Add a ramdisk file to the image
+    pub fn with_ramdisk(&self, ramdisk_path: &Path) -> Self {
+        Self {
+            kernel: self.kernel.to_owned(),
+            ramdisk: Some(ramdisk_path.to_owned()),
         }
     }
```

Comment:
My preferred style of the builder pattern is to take and return `&mut self`:

```suggestion
    /// Add a ramdisk file to the image
    pub fn set_ramdisk(&mut self, ramdisk_path: &Path) -> &mut Self {
        self.ramdisk = Some(ramdisk_path.to_owned());
        self
    }
```

This way, you can use it either through `BiosBoot::new(..).set_ramdisk(..).create_disk_image(..)` or through:

```rust
let mut boot = UefiBoot::new(..);
boot.set_ramdisk(..);
boot.create_disk_image(..);
```